feature/interests/src/main/java/com/google/samples/apps/nowinandroid/feature/interests/InterestsViewModel.kt

```diff
@@ -34,30 +46,83 @@ import javax.inject.Inject
 class InterestsViewModel @Inject constructor(
     val userDataRepository: UserDataRepository,
     getFollowableTopics: GetFollowableTopicsUseCase,
+    userNewsResourceRepository: UserNewsResourceRepository,
+    topicsRepository: TopicsRepository,
+    savedStateHandle: SavedStateHandle,
 ) : ViewModel() {
 
-    val uiState: StateFlow<InterestsUiState> =
-        getFollowableTopics(sortBy = TopicSortField.NAME).map(
-            InterestsUiState::Interests,
-        ).stateIn(
+    private val topicId: String? by lazy {
```

Comment:
I have a sense that topicId can be exposed as a Flow like.

How about the code like below?
```
    private val topicId = savedStateHandle.getStateFlow<String?>(topicIdArg, null)
```

Then for interestsUiState,

```
val interestUiState: StateFlow<InterestsUiState> = topicId.flatMapLatest { topicId ->
            getFollowableTopics(sortBy = TopicSortField.NAME).map {
                InterestsUiState.Interests(
                    topics = it,
                    selectedTopicId = topicId,
                )
            }
        }.stateIn(
            scope = viewModelScope,
            started = SharingStarted.WhileSubscribed(5_000),
            initialValue = InterestsUiState.Loading,
        )
```